xpath-outcome-initiator/src/main/java/org/cristalise/kernel/persistency/outcomebuilder/field/DecimalField.java

```diff
@@ -46,4 +60,122 @@ public void setValue(Object value) throws InvalidOutcomeException {
 
         super.setData(value.toString());
     }
+
+    @Override
+    protected void setAppInfoDynamicFormsExceptionValue(String name, String value) {
+        super.setAppInfoDynamicFormsExceptionValue(name, value);
+
+        if (name.equals("precision")) {
+            precision = value;
+        }
+        else if (name.equals("scale")) {
+            scale = value;
+        }
+    }
+
+    /**
+     * Compute validation regex pattern in this order of precedence:
+     * <pre>
+     * 1. use pattern as is if available
+     * 2. generate pattern from precision and scale if available
+     * 3. generate pattern from totalDigits and fractionDigits if available
+     * 4. set default pattern
+     * </pre>
+     */
+    @Override
+    public void setNgDynamicFormsValidators(JSONObject validators) {
+        super.setNgDynamicFormsValidators(validators);
+
+        if (StringUtils.isNotBlank(pattern)) {
+            validators.put("pattern", pattern);
+        }
+        else if (StringUtils.isNotBlank(precision) || StringUtils.isNotBlank(scale)) {
+            validators.put("pattern", generatePrecisionScalePattern());
+        }
+        else {
+            //this case also generates the default pattern
+            validators.put("pattern", generateTotalFractionDigitsPattern());
+        }
+    }
+
+    private String generatePattern(Integer precisionNumber, boolean precisionSmaller, Integer scaleNumber, boolean scaleSmaller) {
+        //locale specific separators could be used, but it should be based on the locale of the browser
+        //char separator = new DecimalFormatSymbols(Locale.getDefault(Locale.Category.FORMAT)).getDecimalSeparator();
+        char separator = Gateway.getProperties().getString("Webui.decimal.separator", ".").charAt(0);
+
+        if (precisionNumber == null && scaleNumber == null) {
+            //default validator for any decimal field
+            return "^\\d+\\" + separator + "?\\d+$";
+        }
+        else if (precisionNumber != null) {
+            if (scaleNumber == null) {
+                if (precisionSmaller) return "^\\d{0," + precisionNumber + "}$";
+                else                  return "^\\d{" + precisionNumber + "}$";
+            }
+            else {
+                if (precisionSmaller && scaleSmaller)
+                    return "^\\d{0," + (precisionNumber - scaleNumber) + "}\\" + separator + "?\\d{0," + scaleNumber + "}$";

```

Comment:
Here the first part should be \\d{0,precision}, because 1234.5 is acceptable for prec 5 and scale 2. I actually think a regexp cannot cover all cases, we cannot write a regexp that accepts 1234.5 and 123.45, but not 1234.56 - we wopuld need code for such a validation. But this case will be caught by the XSD validation.